Check in/72163/video appointment icon labels

wullaski opened the pull request:
## Summary

- Adds vvc and cvt labels and icons to upcoming appointment list

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#72163](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72163)

## Testing done

- Unit
- Visual

## Screenshots

![Screenshot 2024-05-13 at 10 06 26 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/2982977/0729ae50-6889-4c7a-880b-1941568ce740)

## What areas of the site does it impact?

Check in upcoming appointments list

## Acceptance criteria

- [ ] Content matches wire frames for vvc and ctv appointments.

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

brianseek committed:
init

wullaski committed:
Check in/62227/create upcoming appointments component (#25866)

* init

* Add actions, reducers and selectors for upcoming appointments data

* Add method and mocks to fetch appointments

* Add hook for getting upcoming appointments

* Add components for the list and test

* Add multiple months and days to mocks

* Add sorting

* Add reload

* Fix sort

* style components

* Switch to an array structure for more reliable sorting

* Add error state for upcomingAppointments component

* Fix appointments page test

* Add unit tests for UpcomingAppointments component

* Add test for upcomingAppointmentsList component

* Add test for upcomingAppointmentsListItem component

* Add cypress tests for upcoming appointments

* Fix test

* Change loading messages

* Fix test

* Add i18n for clinic

* Add app to dependency array

* Add dependency

* Add key

* Use i18n utils for date strings

* Fix unit test

* Add translated error message

* Add dependency

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

wullaski committed:
Check in/62138/add what to do next component (#26211)

* init

* Add what to do next component

* Add action item WIP

* Finish up component logic and structure

* Dry upcoming appointments

* Add unit tests for action item display component

* Add what to do next unit test

* Remove eyebrow

* Add e2e tests for day of appointments landing

* Add action link component and unit test

* Test for appointment details click

* Add cypress tests for pre check in

* Add better aria labels and translations

* Remove reloadable

* Add create appointment to better test pre check in card title

* Add pre check in title method

* Fix cypress tests

* Fix unit tests

* Fix unit test

* Deleted a test

* Add button to pass old tests

* Skipping some tests

* Add timeout

* Add cy intercept for mock upcoming appointment data

* Add type button

* remove hard coded string

* change test name for clarity

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

brianseek committed:
Check in/66619/refresh upcoming apts (#26220)

* init

* refactor to keep page from re-fetching data without a relaod.

* removed un-needed param

* Updated tests.

* added test for full hook covereage and removed extra at

* Skipped tests that need un-related refactors.

* Added more test skips.

* data loading refactor for both upcoming and vet data

* Update unit tests.

* removing tests for unified for now :)

These tests will be re-written in upcoming tickets before the feature gets merged to main.

* removed cy screenshots

wullaski committed:
Check in/71267/add review button pre check in details (#27248)

* init

* Updated validate page text. (#25212)

* Check in/62227/create upcoming appointments component (#25866)

* init

* Add actions, reducers and selectors for upcoming appointments data

* Add method and mocks to fetch appointments

* Add hook for getting upcoming appointments

* Add components for the list and test

* Add multiple months and days to mocks

* Add sorting

* Add reload

* Fix sort

* style components

* Switch to an array structure for more reliable sorting

* Add error state for upcomingAppointments component

* Fix appointments page test

* Add unit tests for UpcomingAppointments component

* Add test for upcomingAppointmentsList component

* Add test for upcomingAppointmentsListItem component

* Add cypress tests for upcoming appointments

* Fix test

* Change loading messages

* Fix test

* Add i18n for clinic

* Add app to dependency array

* Add dependency

* Add key

* Use i18n utils for date strings

* Fix unit test

* Add translated error message

* Add dependency

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

* Check in/68865/remove appointment details list view (#26849)

* init

* Check in/62227/create upcoming appointments component (#25866)

* init

* Add actions, reducers and selectors for upcoming appointments data

* Add method and mocks to fetch appointments

* Add hook for getting upcoming appointments

* Add components for the list and test

* Add multiple months and days to mocks

* Add sorting

* Add reload

* Fix sort

* style components

* Switch to an array structure for more reliable sorting

* Add error state for upcomingAppointments component

* Fix appointments page test

* Add unit tests for UpcomingAppointments component

* Add test for upcomingAppointmentsList component

* Add test for upcomingAppointmentsListItem component

* Add cypress tests for upcoming appointments

* Fix test

* Change loading messages

* Fix test

* Add i18n for clinic

* Add app to dependency array

* Add dependency

* Add key

* Use i18n utils for date strings

* Fix unit test

* Add translated error message

* Add dependency

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

* Check in/62138/add what to do next component (#26211)

* init

* Add what to do next component

* Add action item WIP

* Finish up component logic and structure

* Dry upcoming appointments

* Add unit tests for action item display component

* Add what to do next unit test

* Remove eyebrow

* Add e2e tests for day of appointments landing

* Add action link component and unit test

* Test for appointment details click

* Add cypress tests for pre check in

* Add better aria labels and translations

* Remove reloadable

* Add create appointment to better test pre check in card title

* Add pre check in title method

* Fix cypress tests

* Fix unit tests

* Fix unit test

* Deleted a test

* Add button to pass old tests

* Skipping some tests

* Add timeout

* Add cy intercept for mock upcoming appointment data

* Add type button

* remove hard coded string

* change test name for clarity

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

* Check in/66619/refresh upcoming apts (#26220)

* init

* refactor to keep page from re-fetching data without a relaod.

* removed un-needed param

* Updated tests.

* added test for full hook covereage and removed extra at

* Skipped tests that need un-related refactors.

* Added more test skips.

* data loading refactor for both upcoming and vet data

* Update unit tests.

* removing tests for unified for now :)

These tests will be re-written in upcoming tickets before the feature gets merged to main.

* removed cy screenshots

* Add appointment id to form data on action item click

* Remove details view

* change back to appointments link to go back to appointments list view

* Change demographics questions to go to complete page with id

* Fix typo

* Add api calls to complete index

* Remove duplicate method

* Fix unit test

* Delete spaces

* Update and use demographics patch mock response

* Changed use state location and values to prevent calling endpoint twice

* Fixed imports

* WIP add tests for confirmation page

* Add more tests for confirmation component

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

* Check in/71432/unified review page (#27135)

* adjsutments for unified flow.

* Removed re-introduced tests, and added new e2e tests and text for unified.

* Removed details that got back in.

* Add review button to details view

* Remove details link from card when appointments are more than 1

* Add cypress tests of details flow

* Fix details link view logic

* Add upcoming appointments details link click

* Fix what to do next details link logic again final

* Add useStorage and tests

* Delete some tests that got added back in by the rebase

* Remove some duplicate lines that the rebase added

* A few fixes

* Remove unused import

---------

Co-authored-by: Brian Seek <brian.seek@civicactions.com>

brianseek committed:
Check in/70720/refresh button (#29536)

* Added capitalize util.

* Updated page title.

* Also reset complete on refresh.

* Added refresh button and refresh-ability to appointments page.

* added refresh timer.

* no longer testing text for page header.

* updated unit test.

* Added e2e test.

* removed old pages.

* updated upcoming unit test.

wullaski committed:
Check in/71995/modify to use vaos data (#29585)

* Add mock appointment

* Set up mock response

* Change endpoint for upcoming appointments and mocks to match api

* Change hook to recieve api response object

* Convert appointments to match vista appointment object so we can resuse existing shared components and utils

* Modify getAppointmentId to use the VAOS appointment id if it exists and add findUpcomingAppointment method to display the correct VAOS appointment on the appointment details view

* Fix test

* Fix cy intercept

* Skip test for now

* Fix cypress test

* Skipping test

* Add tests and use locationId

wullaski committed:
Add cvt and vvc icons and labels to upcoming list

wullaski committed:
Add kinds to mock data

wullaski committed:
Clean up some bad merges

wullaski committed:
Fix bad merges

brianseek reviewed (commented):
Code looks great! there is a bit of a display issue on my end that I don't seen in your screenshots. Maybe a browser rendering issue.  The icon text alignment is off
<img width="290" alt="Screenshot 2024-05-13 at 1 53 50 PM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/13967174/a534ffe3-1944-497b-9359-1d7aa1a7bb99">
I would suggest using `vads-u-align-items--flex-start` in the container rather than `vads-u-align-items--baseline`
Like so:
<img width="631" alt="Screenshot 2024-05-13 at 1 55 44 PM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/13967174/cad0ba14-0f79-4cde-be2f-505b0b4ec366">

wullaski committed:
Adjust spacing, size and border

wullaski committed:
Add translation for video

wullaski commented:
@brianseek I fixed that icon spacing I needed to do a yarn install to see the problem. I also adjusted some spacing and added borders with the right color.

brianseek commented on src/applications/check-in/components/UpcomingAppointmentsList.jsx:
```diff
@@ -86,19 +86,28 @@ const UpcomingAppointmentsList = props => {
-                              goToDetails={handleDetailClick}
-                              router={router}
-                            />
+                            <div
```
this DIV should go inside the <li> the immediate child of a <ul> should be an <li>

wullaski committed:
Moved border to item component

brianseek commented on src/applications/check-in/components/UpcomingAppointmentsListItem.jsx:
```diff
@@ -97,6 +98,7 @@ const UpcomingAppointmentsListItem = props => {
 
 UpcomingAppointmentsListItem.propTypes = {
   appointment: PropTypes.object.isRequired,
+  border: PropTypes.object.isRequired,
```
this should be `border: PropTypes.bool.isRequired,`

wullaski committed:
fix proptype

brianseek reviewed (approved):
🥇
